src/NodaTime.Test/Calendars/BclWeekYearRuleTest.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright 2016 The Noda Time Authors. All rights reserved.
+// Use of this source code is governed by the Apache License 2.0,
+// as found in the LICENSE.txt file.
+
+using NUnit.Framework;
+using System;
+using System.Globalization;
+using static System.Globalization.CalendarWeekRule;
+using static System.DayOfWeek;
+using NodaTime.Calendars;
+
+namespace NodaTime.Test.Calendars
+{
+    public class BclWeekYearRuleTest
+    {
+        /// <summary>
+        /// For each calendar and rule combination, check everything we can about every date
+        /// from mid December to mid January around each year between 2016 and 2046.
+        /// (For non-Gregorian calendars, the rough equivalent is used...)
+        /// That should give us plenty of coverage.
+        /// </summary>
+        [Test]
+        [Combinatorial]
+        public void BclEquivalence(
+            [ValueSource(typeof(BclCalendars), nameof(BclCalendars.MappedCalendars))] Calendar calendar,
+            [Values(FirstDay, FirstFourDayWeek, FirstFullWeek)] CalendarWeekRule bclRule,
+            [Values(Monday, Tuesday, Wednesday, Thursday, Friday, Saturday, Sunday)]  DayOfWeek firstDayOfWeek)
+        {
+            var nodaCalendar = BclCalendars.CalendarSystemForCalendar(calendar);
+            var nodaRule = BclWeekYearRule.GetRule(bclRule, firstDayOfWeek);
```

Comment:
I think that's just as problematic (which is to say, not really): you have a BCL rule and you want to make a Noda Time rule, but the two things aren't equivalent.

perhaps just s/bclRule/calendarWeekRule/ (per firstDayOfWeek, which isn't called bclDay).